File: FactoryMachines.cpp
#include <algorithm>
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

typedef long long ll;

ll valid(ll t, ll T, const vector<ll> &ks) {
  ll count = 0;
  int n = ks.size();
  for (int i = 0; i < n; i++) {
    count += t / ks[i];
    if (count >= T) {
      return true;
    }
  }
  return false;
}

int main() {
  ll n, t;
  cin >> n >> t;

  vector<ll> ks;

  if (n == 1) {
    ll k;
    cin >> k;
    cout << k * t;
    return 0;
  }

  for (ll i = 0; i < n; i++) {
    ll k;
    cin >> k;
    ks.push_back(k);
  }

  ll lo = 0;
  ll hi = 1e18;
  while (lo <= hi) {
    ll mid = lo + (hi - lo) / 2;
    if (valid(mid, t, ks)) {
      hi = mid - 1;
    } else {
      lo = mid + 1;
    }
  }
  cout << lo;
}

File: factory_machines.py
def total_after_time(t, ks):
    count = 0
    for i in range(len(ks)):
        count += t // ks[i]
    return count


def main():
    n, t = [int(y) for y in input().split()]
    ks = [int(y) for y in input().split()]

    if n == 1:
        print(ks[0] * t)
        return

    lo = 0
    hi = 1e18
    while lo <= hi:
        mid = lo + (hi - lo) // 2
        if total_after_time(mid, ks) >= t:
            hi = mid - 1
        else:
            lo = mid + 1

    print(int(lo))


main()
